packages/compass-e2e-tests/tests/collection-import.test.ts

```diff
@@ -35,39 +112,186 @@ describe('Collection import', function () {
     expect(await insertConfirm.isDisplayed()).to.be.true;
     expect(await insertConfirm.getText()).to.equal('Insert');
     await insertConfirm.waitForEnabled();
     await browser.clickVisible(Selectors.InsertConfirm);
 
+    // wait for the modal to go away
     await insertDialog.waitForDisplayed({ reverse: true });
+
+    // make sure the documents appear in the collection
     const messageElement = await browser.$(
       Selectors.DocumentListActionBarMessage
     );
     await browser.waitUntil(async () => {
       const text = await messageElement.getText();
       return text === 'Displaying documents 1 - 20 of 1000';
     });
+
+    const result = await getFirstListDocument(browser);
+
+    expect(result._id).to.exist;
+    delete result._id;
+
+    expect(result).to.deep.equal({
+      n: '0',
+      n_square: '0',
+    });
+  });
+
+  it('displays an error for a malformed JSON array', async function () {
+    await browser.navigateToCollectionTab('test', 'json-array', 'Documents');
+
+    const json = 'this is not valid JSON';
+
+    // browse to the "Insert to Collection" modal
+    await browser.clickVisible(Selectors.AddDataButton);
+    const insertDocumentOption = await browser.$(
+      Selectors.InsertDocumentOption
+    );
+    await insertDocumentOption.waitForDisplayed();
+    await browser.clickVisible(Selectors.InsertDocumentOption);
+
+    // wait for the modal to appear
+    const insertDialog = await browser.$(Selectors.InsertDialog);
+    await insertDialog.waitForDisplayed();
+
+    // set the text in the editor
+    await browser.setAceValue(Selectors.InsertJSONEditor, json);
+
+    // make sure that there's an error and that the insert button is disabled
+    const errorElement = await browser.$(Selectors.InsertDialogErrorMessage);
+    await errorElement.waitForDisplayed();
+    expect(await errorElement.getText()).to.equal(
+      'Insert not permitted while document contains errors.'
+    );
+    const insertButton = await browser.$(Selectors.InsertConfirm);
+    await insertButton.waitForEnabled({ reverse: true });
+
+    // cancel and wait for the modal to go away
+    await browser.clickVisible(Selectors.InsertCancel);
+    await insertDialog.waitForDisplayed({ reverse: true });
   });
 
   it('supports JSON files', async function () {
     const jsonPath = path.resolve(__dirname, '..', 'fixtures', 'listings.json');
 
     await browser.navigateToCollectionTab('test', 'json-file', 'Documents');
 
+    await importJSONFile(browser, jsonPath);
+
+    const messageElement = await browser.$(
+      Selectors.DocumentListActionBarMessage
+    );
+    const text = await messageElement.getText();
+    expect(text).to.equal('Displaying documents 1 - 20 of 16116');
+
+    const result = await getFirstListDocument(browser);
+
+    expect(result._id).to.exist;
+    delete result._id;
+
+    expect(result).to.deep.equal({
+      id: '2818',
+      name: '"Quiet Garden View Room & Super Fast WiFi"',
```

Comment:
I had somehow always assumed that our test data was actual test data or at least redacted in some way, but … [I guess not](https://www.airbnb.com/rooms/2818?source_impression_id=p3_1645013842_sXWykOdGbU2GwVgM) :)

(not an issue, just something curious)